core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala

```diff
@@ -34,51 +32,12 @@ import reactor.core.publisher.Mono
  */
 @InternalStableApi object R2dbcExecutor {
   final implicit class PublisherOps[T](val publisher: Publisher[T]) extends AnyVal {
-    def asFuture(): Future[T] = {
-      val promise = Promise[T]()
-      publisher.subscribe(new Subscriber[T] {
-
-        override def onSubscribe(s: Subscription): Unit = {
-          s.request(1)
-        }
-
-        override def onNext(value: T): Unit = {
-          promise.trySuccess(value)
-        }
-
-        override def onError(t: Throwable): Unit = {
-          promise.tryFailure(t)
-        }
-
-        override def onComplete(): Unit = {
-          if (!promise.isCompleted)
-            promise.tryFailure(new RuntimeException(s"Publisher [$publisher] completed without first value."))
-        }
-      })
-      promise.future
-    }
+    def asFuture(): Future[T] =
+      Mono.from(publisher).toFuture.toScala
 
     def asFutureDone(): Future[Done] = {
-      val promise = Promise[Done]()
-      publisher.subscribe(new Subscriber[Any] {
-
-        override def onSubscribe(s: Subscription): Unit = {
-          s.request(1)
-        }
-
-        override def onNext(value: Any): Unit = {
-          promise.trySuccess(Done)
-        }
-
-        override def onError(t: Throwable): Unit = {
-          promise.tryFailure(t)
-        }
-
-        override def onComplete(): Unit = {
-          promise.trySuccess(Done)
-        }
-      })
-      promise.future
+      val mono: Mono[Done] = Mono.from(publisher).map(_ => Done)
+      mono.defaultIfEmpty(Done).toFuture.toScala
     }
   }
```

Comment:
Tested performance with R2dbcJournalPerfSpec and R2dbcJournalPerfManyActorsSpec and local Postgres. No significant difference.

NEW:
```
[info] A PersistentActor's performance
[info] - must measure: persistAsync()-ing 2000 events (12 seconds, 369 milliseconds)
[info]   + PersistAsync()-ing 2000 took 2501 ms
[info]   + PersistAsync()-ing 2000 took 2132 ms
[info]   + PersistAsync()-ing 2000 took 1899 ms
[info]   + PersistAsync()-ing 2000 took 1942 ms
[info]   + PersistAsync()-ing 2000 took 1955 ms
[info]   + PersistAsync()-ing 2000 took 1913 ms
[info]   + Average time: 2057 ms
[info] - must measure: persist()-ing 2000 events (20 seconds, 94 milliseconds)
[info]   + Persist()-ing 2000 took 4093 ms
[info]   + Persist()-ing 2000 took 3698 ms
[info]   + Persist()-ing 2000 took 3107 ms
[info]   + Persist()-ing 2000 took 3137 ms
[info]   + Persist()-ing 2000 took 3084 ms
[info]   + Persist()-ing 2000 took 2970 ms
[info]   + Average time: 3348 ms
[info] - must measure: recovering 2000 events (3 seconds, 537 milliseconds)
[info]   + Recovering 2000 took 69 ms
[info]   + Recovering 2000 took 45 ms
[info]   + Recovering 2000 took 37 ms
[info]   + Recovering 2000 took 36 ms
[info]   + Recovering 2000 took 36 ms
[info]   + Recovering 2000 took 33 ms
[info]   + Average time: 43 ms

[info] A PersistentActor's performance
[info] - must measure: persist()-ing 200 events for 100 actors (21 seconds, 786 milliseconds)
[info]   + Persist()-ing 200 * 100 took 4254 ms
[info]   + Persist()-ing 200 * 100 took 3633 ms
[info]   + Persist()-ing 200 * 100 took 3505 ms
[info]   + Persist()-ing 200 * 100 took 3378 ms
[info]   + Persist()-ing 200 * 100 took 3397 ms
[info]   + Persist()-ing 200 * 100 took 3551 ms
[info]   + Average time: 3620 ms
```

OLD:
```
[info] A PersistentActor's performance
[info] - must measure: persistAsync()-ing 2000 events (13 seconds, 54 milliseconds)
[info]   + PersistAsync()-ing 2000 took 2636 ms
[info]   + PersistAsync()-ing 2000 took 2031 ms
[info]   + PersistAsync()-ing 2000 took 2187 ms
[info]   + PersistAsync()-ing 2000 took 2087 ms
[info]   + PersistAsync()-ing 2000 took 2012 ms
[info]   + PersistAsync()-ing 2000 took 2075 ms
[info]   + Average time: 2171 ms
[info] - must measure: persist()-ing 2000 events (20 seconds, 272 milliseconds)
[info]   + Persist()-ing 2000 took 3996 ms
[info]   + Persist()-ing 2000 took 3425 ms
[info]   + Persist()-ing 2000 took 3297 ms
[info]   + Persist()-ing 2000 took 3138 ms
[info]   + Persist()-ing 2000 took 3164 ms
[info]   + Persist()-ing 2000 took 3247 ms
[info]   + Average time: 3378 ms
[info] - must measure: recovering 2000 events (3 seconds, 506 milliseconds)
[info]   + Recovering 2000 took 82 ms
[info]   + Recovering 2000 took 43 ms
[info]   + Recovering 2000 took 39 ms
[info]   + Recovering 2000 took 37 ms
[info]   + Recovering 2000 took 36 ms
[info]   + Recovering 2000 took 35 ms
[info]   + Average time: 45 ms

[info] A PersistentActor's performance
[info] - must measure: persist()-ing 200 events for 100 actors (22 seconds, 111 milliseconds)
[info]   + Persist()-ing 200 * 100 took 4226 ms
[info]   + Persist()-ing 200 * 100 took 3777 ms
[info]   + Persist()-ing 200 * 100 took 3168 ms
[info]   + Persist()-ing 200 * 100 took 3290 ms
[info]   + Persist()-ing 200 * 100 took 3330 ms
[info]   + Persist()-ing 200 * 100 took 4272 ms
[info]   + Average time: 3677 ms
```